mail/postfix/Makefile

```diff
@@ -177,136 +181,78 @@ newaliases_path=/usr/bin/newaliases
 mailq_path=/usr/bin/mailq
 shlib_directory=/usr/lib/postfix
 meta_directory=/etc/postfix
+# postinit script may override the values below if the external
+# drive is at a mountpoint other than /data/
+data_directory=/data/postfix/lib
+queue_directory=/data/postfix/spool
+mail_spool_directory=/data/postfix/mail
 
 ln_suffix=.postfix
-ln_old_suffix=.old
+ln_suffix_pre_pf=.pre-postfix
 
 define Package/postfix/conffiles
-$(config_directory)/main.cf
-$(config_directory)/master.cf
-$(config_directory)/aliases
-$(config_directory)/virtual
+/etc/postfix/
```

Comment:
You are correct, the binary files would also be included and it would increase the size of sysupgrade conffiles, therefore raising the potential for sysupgrade to fail.

However, the compiled db or cdb files in /etc/postfix/ add very little to the combined size of all source configuration files. In my case, "du -sh /etc/postfix" tells me that /etc/postfix takes up 380KB on my production server and 280KB on my development server. "aliases.cdb" and "virtual.cdb" together occupy about 6KB, which is 1.6%-2.1% of the combined size of /etc/postfix/.

IMO the downside of including the compiled binary files in conffiles is rather small. It increases the size of conffiles by about 2%. However, the benefit of including all configuration files and not just main.cf, master.cf, aliases and virtual is significant. At present, all configuration files other than those four are guaranteed to be lost on sysupgrade.

There are other configuration files that are often used, such as: access, canonical, generic, header_checks, relocated and transport. There may be other configuration files with arbitrary names. In my case, I also use controlled_envelope_senders, header_checks.pcre, ldap_controlled_envelope_senders.cf and ldap_virtual_mb_maps.cf. All these configuration files are guaranteed to be lost on sysupgrade at present (unless extroot is used).

If there were a method to **exclude** certain files within /etc/postfix, I would be happy to use that mechanism to exclude *.db and *.cdb files. AFAIK there is no such method.

Reply:
Ok, I agree, but let's change /etc/postfix to $(config_directory)